src/component/Greeting.js

```diff
@@ -0,0 +1,23 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchGreetingApi } from '../redux/greetings/greetings';
+
+const Greeting = () => {
+  const greeting = useSelector((state) => state.greetingsReducer);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchGreetingApi());
+  }, [dispatch]);
+
+  const refreshPage = () => {
+    window.location.reload(false);
+  };
```

Comment:
- [ ] <b>OPTIONAL:</b> You don't have to refresh the page. if your want to add refresh functionality. you can use rerender instead.
   - First, rename `refreshPage ` function to `handleRefresh` and change it like so.
     ```js
        const handleRefresh  = () => {
           dispatch(fetchGreetingApi());
        }
     ```
    - Then import shallowEqual from `react-redux`
     ```js
        import { shallowEqual, useSelector } from 'react-redux'
    ```
    - And the you use it like this
    ```js
       const greeting = useSelector((state) => state.greetingsReducer,  shallowEqual);
       
      // and also you set onClick of your function to handleRefresh
       <button type="button" onClick={handleRefresh}>Get another greeting</button>

    ```